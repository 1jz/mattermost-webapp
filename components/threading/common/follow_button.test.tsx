// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {mountWithIntl} from 'tests/helpers/intl-test-helper';

import FollowButton from './follow_button';
import Button from './button';

describe('components/threading/common/follow_button', () => {
    test('should say follow and fire start action', () => {
        const start = jest.fn();
        const stop = jest.fn();

        const wrapper = mountWithIntl(
            <FollowButton
                isFollowing={false}
                start={start}
                stop={stop}
            />,
        );

        expect(wrapper).toMatchSnapshot();

        expect(wrapper.find(Button).text()).toBe('Follow');

        wrapper.find(Button).simulate('click');
        expect(start).toHaveBeenCalled();
        expect(stop).not.toHaveBeenCalled();
    });

    test('should say following and fire stop action', () => {
        const start = jest.fn();
        const stop = jest.fn();

        const wrapper = mountWithIntl(
            <FollowButton
                isFollowing={true}
                start={start}
                stop={stop}
            />,
        );

        expect(wrapper).toMatchSnapshot();

        expect(wrapper.find(Button).text()).toBe('Following');

        wrapper.find(Button).simulate('click');
        expect(start).not.toHaveBeenCalled();
        expect(stop).toHaveBeenCalled();
    });
});
