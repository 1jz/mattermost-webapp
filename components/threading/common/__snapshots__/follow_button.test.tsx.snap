// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/threading/common/follow_button should say follow and fire start action 1`] = `
<Memo(FollowButton)
  isFollowing={false}
  start={[MockFunction]}
  stop={[MockFunction]}
>
  <Memo(Button)
    isActive={false}
    onClick={[MockFunction]}
  >
    <button
      className="Button Button___transparent"
      onClick={[MockFunction]}
    >
      <span
        className="Button_label"
      >
        <FormattedMessage
          defaultMessage="Follow"
          id="threading.footer.notFollowing"
        >
          <span>
            Follow
          </span>
        </FormattedMessage>
      </span>
    </button>
  </Memo(Button)>
</Memo(FollowButton)>
`;

exports[`components/threading/common/follow_button should say following and fire stop action 1`] = `
<Memo(FollowButton)
  isFollowing={true}
  start={[MockFunction]}
  stop={[MockFunction]}
>
  <Memo(Button)
    isActive={true}
    onClick={[MockFunction]}
  >
    <button
      className="Button Button___transparent is-active"
      onClick={[MockFunction]}
    >
      <span
        className="Button_label"
      >
        <FormattedMessage
          defaultMessage="Following"
          id="threading.footer.following"
        >
          <span>
            Following
          </span>
        </FormattedMessage>
      </span>
    </button>
  </Memo(Button)>
</Memo(FollowButton)>
`;
